### PR TITLE
Add tests for the ipaddr filter

### PR DESCRIPTION
**Pull request by mscherer**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

test
##### SUMMARY

Here is a first list of test on the ipaddr filter. I didn't add all yet, as I am not sure which one are useful to test and which aren't, and what could be corner cases to test.

Also, tests will fail until #17627 is merged.

### CODE REVIEW

**Comment by alikins** [comment]
Seems to be failing because integration test systems don't have python-netaddr

```
The error was: The ipaddr filter requires python-netaddr be installed on the ansible controller"
```

I'm not entirely sure what the best place to add that requirement is though.

ansible/test/utils/shippable/integration-requirements.txt  ?
ansible/test/utils/tox/requirements.txt

**Comment by alikins** [comment]
(needs_revision because of the python-netaddr issue)

**Comment by mscherer** [comment]
I would place it in ansible/test/utils/shippable/integration-requirements.txt, but @mattclay is the one who will know.

**Comment by mattclay** [comment]
Yes, adding it to `test/utils/shippable/integration-requirements.txt` is correct.

**Comment by calfonso** [comment]
@mscherer would you mind rebasing this please, thanks!

**Comment by gundalow** [comment]
ipaddr can be added to `test/utils/shippable/integration-requirements.txt`

**Comment by mattclay** [comment]
CI failure in integration tests. Here's one example for ```test/integration/targets/filters/tasks/main.yml:129 / [testhost] testhost: filters : Test ipaddr filter that=[u"'192.168.0.1/32' | ipaddr('netmask') == '255.255.255.255'", u"'192.168.0.1/24' | ipaddr('netmask') == '255.255.255.0'", u"'192.168.0.1/32' | ipaddr('broadcast') is sameas false", u"'192.168.0.1/24' | ipaddr('broadcast') == '192.168.0.255'", u"'192.168.0.1/24' | ipaddr('prefix') == '24'", u"'192.168.0.1/24' | ipaddr('address') == '24'", u"'192.168.0.1/24' | ipaddr('network') == '192.168.0.0'", u"'::1/120' | ipaddr('broadcast') == '::ff'", u"'::1/120' | ipaddr('netmask') == 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ff00'"]```:

```
{
    "assertion": "'192.168.0.1/32' | ipaddr('broadcast') is sameas false", 
    "changed": false, 
    "evaluated_to": false, 
    "failed": true
}    
```

File: test/runner/requirements/integration.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ passlib
 pexpect
 pycrypto
 pyyaml
+netaddr
```
Comment (mattclay) [inline test/runner/requirements/integration.txt:10]: I'm revising my previous suggestion to add this to the requirements here. Instead, please use the pip module to install the requirement at the start of the test. That way we don't install test-specific dependencies when not running those tests.

**Comment by mattclay** [comment]
@mscherer Now that #17627 is merged, can you update this PR so tests can run?

**Comment by Akasurde** [comment]
@mscherer Are you still working on this ? Let me know if not.

**Comment by Akasurde** [comment]
superseded by #52088

**Pull request closed without merging**